# Random Forest Classifier

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import train_test_split

from sklearn.metrics import accuracy_score, f1_score, precision_score

In [8]:
# Get labels from kmeans_pca_3.csv
labels = pd.read_csv('kmeans_pca_3.csv')
labels = labels.loc[:, 'Labels']


In [14]:
# Get features from pca_3.csv (3 features) and pca_80(7 features)
feats_3 = pd.read_csv('pca_3.csv')
feats_3 = feats_3.drop('Unnamed: 0', axis=1)

feats_7 = pd.read_csv('pca_80.csv')
feats_7 = feats_7.drop('Unnamed: 0', axis=1)

# Get training and Testing Splits (80/20)
x_3_train, x_3_test, y_3_train, y_3_test = train_test_split(feats_3, labels, test_size=0.2) # TODO: Could try default test_size
x_7_train, x_7_test, y_7_train, y_7_test = train_test_split(feats_7, labels, test_size=0.2) # TODO: Could try default test_size

# TODO: Possibly need numpy conversions for y_test values

In [44]:
# TODO: Run model on both sets of features
clf = RandomForestClassifier(n_estimators=95, max_depth=15, random_state=0) # TODO: Parameter tuning, might fail without setting max_depth and some other parameters
clf.fit(x_3_train, y_3_train) # Fit the model
yh = clf.predict(x_3_test)
#print("Predicted: ", yh)
#print("Actual: ", y_3_test)

clf_7 = RandomForestClassifier(n_estimators=128, max_depth=None, random_state=0) # TODO: Parameter tuning, might fail without setting max_depth and some other parameters
clf_7.fit(x_7_train, y_7_train) # Fit the model
yh_7 = clf_7.predict(x_7_test)
#print("Predicted: ", yh_7)
#print("Actual: ", y_7_test)

In [45]:
# TODO: Find accuracy, precision, and f1 score

print("-----------------3 features-----------------")

accuracy = accuracy_score(y_3_test, yh)
print("Accuracy: ", accuracy)

precision = precision_score(y_3_test, yh, average="weighted")
print("Precision: ", precision)

f1 = f1_score(y_3_test, yh, average='weighted')
print("F1 Score: ", f1)

print("-----------------7 features-----------------")

accuracy = accuracy_score(y_7_test, yh_7)
print("Accuracy: ", accuracy)

precision = precision_score(y_7_test, yh_7, average="weighted")
print("Precision: ", precision)

f1 = f1_score(y_7_test, yh_7, average='weighted')
print("F1 Score: ", f1)


-----------------3 features-----------------
Accuracy:  0.9908038851002273
Precision:  0.9908059991028073
F1 Score:  0.9908040343346386
-----------------7 features-----------------
Accuracy:  0.9831576772060343
Precision:  0.9831548262386596
F1 Score:  0.9831545526152303


```python:
max_depth=None, random_state=0
-----------------3 features-----------------
Accuracy:  0.990700557966522
Precision:  0.9907095866611935
F1 Score:  0.990703084542843
-----------------7 features-----------------
Accuracy:  0.9826410415375078
Precision:  0.9826372310723576
F1 Score:  0.9826368790190678

max_depth=15, random_state=0
-----------------3 features-----------------
Accuracy:  0.9908038851002273
Precision:  0.9908053192464386
F1 Score:  0.9908037607517728
-----------------7 features-----------------
Accuracy:  0.9815044430667493
Precision:  0.9814947522383299
F1 Score:  0.9814948636682381

n_estimators=95,128, max_depth=15, random_state=0
-----------------3 features-----------------
Accuracy:  0.9908038851002273
Precision:  0.9908059991028073
F1 Score:  0.9908040343346386
-----------------7 features-----------------
Accuracy:  0.9831576772060343
Precision:  0.9831548262386596
F1 Score:  0.9831545526152303